src/runtime/cuda.cpp

```diff
@@ -104,8 +104,7 @@ WEAK int halide_cuda_acquire_context(void *user_context, CUcontext *ctx, bool cr
     // If the context has not been initialized, initialize it now.
     halide_assert(user_context, &context != NULL);
 
-    CUcontext local_val;
-    __atomic_load(&context, &local_val, __ATOMIC_ACQUIRE);
+    CUcontext local_val = context;
```

Comment:
We should add comments here and ho halide_cuda_device_release explaining that the client must guarantee mutual exclusion between the two. We should also add this documentation to halide_device_release in HalideRuntime.h and likely a TODO that we need to come up with something better. With those doc changes and the PR going green on the buildbots, I think it can be submitted. Does this get rid of all __atomic usage in the file? If not, we should consider doing so per the issues I ran into on 32-bit with fast sync.